test/junitTrace/JupiterTempDir.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledIfSystemProperty;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/*
+ * @test
+ * @bug 7903953
+ * @run junit JupiterTempDir
+ */
+class JupiterTempDir {
+    @Test
+    @EnabledIfSystemProperty(named = "test.name", matches = "JupiterTempDir.java")
```

Comment:
Hello Christian, why is this needed in this test?

Reply:
The assertion(s) in the test only make sense, when `jtreg` runs the test via its `JUnitRunner`. The `test.name` system property is an indicator that `jtreg` runs the test, I could switch to another one documented here: https://openjdk.org/jtreg/tag-spec.html#testvars

Reply:
Perhaps it would be better to replace this non-obvious declarative guard with a programmatic `Assumptions....` statement within the test method. 🤔

Reply:
> The assertion(s) in the test only make sense, when jtreg runs the test via its JUnitRunner.

Since this is a self test in the jtreg project, would it be the case that a test marked as `@run junit` will always be run by the `JUnitRunner`? So even the assumption I think is probably isn't needed.

Reply:
When running this self-test without jtreg in play, for example via a direct launch within an IDE(A), it would fail hard. With the assumption in place, developers see:

```
org.opentest4j.TestAbortedException: Assumption failed: jtreg not running

	at org.junit.jupiter.api.Assumptions.throwAssumptionFailed(Assumptions.java:316)
	at org.junit.jupiter.api.Assumptions.assumeTrue(Assumptions.java:115)
	at JupiterTempDir.currentWorkingDirectoryIsParentOfTemporary(JupiterTempDir.java:44)
```

Adding a comment, as suggested in another channel, will help future folk though.